src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py

```diff
@@ -0,0 +1,199 @@
+                    endpoint,
+                    query_param,
+                    headers=headers
+                    )
+        data = response.json()
+        if data and data.get("message", "").lower() == "success.":
+            response_json = data.get("data", None)
+            break
+        else:
+            response_json = None
+    return response_json
+
+
+def _process_objects_batch(objects_batch):
+    image_count = None
+    for object_ in objects_batch:
+        rights_info = object_.get("rights_type")
+        license_url = _get_license_url(rights_info)
+        logger.debug(license_url)
+        if license_url is not None:
+            id_ = object_.get("id", "")
+            complete_object_data = _get_object_json(
+                endpoint=ENDPOINT+str(id_)
+                )
+            if complete_object_data is None:
+                continue
+            image_count = _handle_object_data(
+                data=complete_object_data,
+                license_url=license_url
+                )
+
+    return image_count
+
+
+def _handle_object_data(data, license_url):
+    image_count = None
+    image_info = data.get("images", None)
+    if image_info is not None:
+        id_ = data.get("id", "")
+        title = data.get("title", "")
+        foreign_url = f"https://www.brooklynmuseum.org/opencollection/objects/{id_}"
+        metadata = _get_metadata(data)
+        creators = _get_creators(data)
+
+        for image in image_info:
+            foreign_id = image.get("id", "")
+            image_url, thumbnail_url = _get_images(image)
+            if image_url is None:
+                continue
+            height, width = _get_image_sizes(image)
+
+            image_count = image_store.add_item(
+                    foreign_landing_url=foreign_url,
+                    image_url=image_url,
+                    license_url=license_url,
+                    foreign_identifier=foreign_id,
+                    width=width,
+                    height=height,
+                    title=title,
+                    thumbnail_url=thumbnail_url,
+                    meta_data=metadata,
+                    creator=creators
+                )
+
+    return image_count
+
+
+def _get_image_sizes(image):
+    height, width = None, None
+    size_list = image.get("derivatives", "")
+    if type(size_list) is list:
+        size_type = image.get("largest_derivative", "")
+        for size in size_list:
+            if size.get("size", "") == size_type:
+                height = size.get("height", None)
+                width = size.get("width", None)
+    return height, width
+
+
+def _get_license_url(rights_info):
+    if "creative commons" in rights_info.get("name").lower():
+        elements = html.fromstring(rights_info.get("description", ""))
+        cc_links = [
+            elm[2]
+            for elm in elements.iterlinks()
+            if "https://creativecommons.org/licenses/" in elm[2]
+        ]
```

Comment:
@mathemancer 
I had gone through all the objects in brooklyn and there is only one license type. 
```
+-----------------+---------------------+---------+---------+
| provider        | license_type        | license | version |
+-----------------+---------------------+---------+---------+
| brooklyn museum | creative commons 3d | by      | 3.0     |
+-----------------+---------------------+---------+---------+

```
So in their JSON the ```rights_type``` has different rights name possible currently we are only checking for ``` creative commons``` , this would miss types with name such as ```public domain ``` .(currently there isn't one but in case )
Instead we could use 
```
def _get_license_url(rights_info):
    elements = html.fromstring(rights_info.get("description", ""))
    cc_links = [
        elm[2]
        for elm in elements.iterlinks()
        if "https://creativecommons.org/" in elm[2]
    ]
    if len(cc_links) == 1:
        (license_url,) = cc_links
    else:
        license_url = None
    return license_url
```